crypto/asn1/bio_asn1.c

```diff
@@ -138,6 +138,13 @@ static int asn1_bio_free(BIO *b)
     if (ctx == NULL)
         return 0;
 
+    if (ctx->prefix_free) {
```

Comment:
Coding style
````
if (ctx->prefix_free != NULL)
    ctx->prefix_free(b, &ctx->ex_buf, &ctx->ex_len, &ctx->ex_arg);
````

Reply:
I would really like to see a test for this..
Especially if ctx->ex_buf, ctx->ex_len, ctx->ex_arg are being shared between the prefix and suffix.

Reply:
@slontis about the test, can I just extend the cms app with an flag to fail reading from the input?

Reply:
A low level test added to bio_memleak_test .c would be sufficient..

Reply:
If you already had some data you should be able to debug into it to see what needs to be set up for a simpler test.

Reply:
Ok, I've added a small unit test which checks for the case. If you comment the "ctx->prefix_free" calls which the patch added, you can verify the memory leak.